src/librustc_metadata/creader.rs

```diff
@@ -854,21 +854,45 @@ impl<'a> CrateLoader<'a> {
                 return
             }
 
-            if !self.sess.crate_types.borrow().iter().all(|ct| {
-                match *ct {
-                    // Link the runtime
-                    config::CrateTypeExecutable => true,
-                    // This crate will be compiled with the required
-                    // instrumentation pass
-                    config::CrateTypeRlib => false,
-                    _ => {
-                        self.sess.err(&format!("Only executables and rlibs can be \
-                                                compiled with `-Z sanitizer`"));
-                        false
+            // The gnu linux target has better support for asan with multiple lib types.
```

Comment:
Is there somewhere we can link to for this comment? I'd personally find it sort of odd that the logic here differs depending on the target just because we haven't been able to figure it out in one scenario.